api/pkg/gateway/context_test.go

```diff
@@ -0,0 +1,541 @@
+package gateway
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/labstack/echo/v4"
+
+	"github.com/shellhub-io/shellhub/pkg/api/authorizer"
+	"github.com/shellhub-io/shellhub/pkg/models"
+	"github.com/stretchr/testify/require"
+)
+
+func TestRole(t *testing.T) {
+	type Expected struct {
+		roleString authorizer.Role
+	}
+
+	cases := []struct {
+		description string
+		headers     map[string]string
+		expected    Expected
+	}{
+		{
+			description: "verify if given value returns from header",
+			headers: map[string]string{
+				"X-Role": "owner",
+			},
+			expected: Expected{
+				roleString: authorizer.Role("owner"),
+			},
+		}, {
+			description: "validate empty ID string, for ID behaviour",
+			headers: map[string]string{
+				"X-Role": "",
+			},
+			expected: Expected{
+				roleString: authorizer.Role(""),
+			},
+		}, {
+			description: "validate empty headers, for fail function",
+			headers: map[string]string{
+				"": "",
+			},
+			expected: Expected{
+				roleString: authorizer.Role(""),
+			},
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.description, func(t *testing.T) {
+			e := echo.New()
+			req := httptest.NewRequest(http.MethodPost, "/", nil)
+
+			req.Header.Set(echo.HeaderContentType, echo.MIMEApplicationJSON)
```

Comment:
I believe that since you're not sending a body, there's no need to set the `Content-Type` header.

Reply:
Understood

Reply:
and a very good one